---
title: "eDNA Experiment Notes"
description: |
  Stickleback experiment notes
author:
  - name: R. Peek 
date: "Updated `r Sys.Date()`"
output: 
  distill::distill_article:
    toc: true
    code_folding: false
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, tidy = FALSE,
                      message = FALSE, warning = FALSE)

library(knitr)
library(here)
library(glue)
```

# Overview

General notes about process and steps to test eDNA population genomics approaches.

## 2022-06-03

Working on `snakemake` approach for downloading the stickleback genome, searching for `sbf1` cut sites, identifying flanking regions, and selecting baits from these 120bp regions.

### Stickleback Genome

Has been mapped, see: 

 - [NCBI FTP Website here](https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/180/675/GCA_000180675.1_ASM18067v1/) 
 - [GenBank site here](https://www.ncbi.nlm.nih.gov/assembly/GCA_000180675.1/)


### Thinking about Tags

Use existing tags or SNPs to select things that may not accidentally select for microbial data or other things (algae, trees, etc). Sourmash would tell you for microbial data, but may need to use BLAST to see more broadly.

 - https://www.ncbi.nlm.nih.gov/sra/SRX015871%5Baccn
 - https://www.ncbi.nlm.nih.gov/sra/SRX028026%5Baccn
 - https://www.ncbi.nlm.nih.gov/assembly/GCA_000180675.1/
 
 
## Try using SRA-tools

You can have a config.yaml that has this info:
```
srr:
  - SRR12345678

```
Then run with the following:

```
configfile: "include/rules/config.yaml"
 
include: "include/rules/fasterqdump.rule"
 
rule all:
    input:
        expand("01_raw/done__{srr}_dump", srr=config['srr'])

rule prefetch:
    output:
        "01_raw/.prefetch/sra/{srr}.sra"
    params:
        "{srr} --max-size 50GB -O 01_raw"
    log:
        "01_raw/.prefetch/sra/{srr}.log"
    conda:
        "yamls/sra-tools.yaml"
    shell:
        """
        prefetch {params} > {log} 2>&1 && touch {output}
        """
 
rule fastqdump:
    input:
        "01_raw/.prefetch/sra/{srr}.sra"
    output:
        touch("01_raw/done__{srr}_dump")
    params:
        args = "-S -O 01_raw/ -t 01_raw/",
        id_srr = "{srr}"
    log:
        "01_raw/{srr}.log"
    conda:
        "yamls/sra-tools.yaml"
    shell:
        """
        fasterq-dump {params.args} {params.id_srr} > {log} 2>&1 
        """
```


Test it out: `snakemake -p -j 30 --use-conda -n`

## Grep for sbf1 (`CCTGCAGG`)

Common [6 bp cutter](https://en.wikipedia.org/wiki/List_of_restriction_enzyme_cutting_sites:_S)

### Using `seqkit`

[Seqkit](https://bioinf.shenwei.me/seqkit/download/)

Get some stats about the file:

```
zcat GCA_000180675.11_67v1_genomic.fna.gz | seqkit stats

# file  format  type  num_seqs      sum_len  min_len   avg_len  max_len
# -     FASTA   DNA     16,957  446,610,788      101  26,337.8  698,234
```
Count Chromosome/Seqs:

```
seqkit seq GCA_0001806ASM18067v1_genomic.fna.gz -n | wc -l
```

Get just the ID:

```
seqkit seq GCA_0001806ASM18067v1_genomic.fna.gz -n -i | head
```

Just print seq:

```
seqkit seq hairpin.fa.gz -s -w 0

```

Grep for Seq containing `CCTGCAGG`:

```
seqkit grep -s -i -p cctgcagg GCA_000180675.11_ASM18067v1_genomic.fna.gz -C | head

# n=7367

# grep and then pull the chrom only:
seqkit grep -s -i -p cctgcagg GCA_000180675.11_ASM18067v1_genomic.fna.gz | seqkit seq -n -i | head
# AANH01000001.1
# AANH01000002.1
# AANH01000003.1
# AANH01000006.1
```

```
seqkit grep -s -i -p cctgcagg GCA_000180675.11_ASM18067v1_genomic.fna.gz | seqkit seq -n -s -l | less
```

Then take that output and remove line breaks and then pipe to pull just the pieces we need (see below)

```
seqkit grep -s -i -p cctgcagg input_assembly/GCA_000180675.11_ASM18067v1_genomic.fna.gz | seqkit seq -n -s -u -w 0 -o output/bait_sbf1_seqs_to_clip.fa.gz
```




### Using straight `grep`

For characters instead of lines, this returns seq only w 120 bp after the cut site (CCTGCAGG):

```
grep -o -P '.{0,0}CCTGCAGG.{0,120}' GCA_000180675.11_ASM18067v1.fq | wc -l
```

# Run

```
snakemake -r -n -c1
```
